server/controllers/finance/indicator/dashboard/index.js

```diff
@@ -0,0 +1,186 @@
+      FROM hospitalization_indicator hi  
+      JOIN indicator ind ON ind.uuid = hi.indicator_uuid
+      JOIN period p ON p.id = ind.period_id
+    `;
+
+    db.convert(options, ['uuid', 'indicator_uuid']);
+
+    // this first filter will be used for the sums query
+    let filters1 = new FilterParser(options, { tableAlias : 'hi' });
+
+    // this second filter will be used for the last value query
+    let filters2 = new FilterParser(options, { tableAlias : 'hi', limitKey : 1 });
+
+    filters1 = defaultFilters(filters1);
+    filters1.equals('service_id');
+
+    filters2 = defaultFilters(filters2);
+    filters2.equals('service_id');
+    filters2.setOrder('ORDER BY p.start_date DESC');
+
+    // groupByPeriod key specify if it is necessary to group by period
+    if (options.groupByPeriod) {
+      filters1.setGroup('GROUP BY p.id');
+      filters2.setGroup('GROUP BY p.id');
+    }
+    // sums query
+    const data1 = await db.exec(filters1.applyQuery(sql1), filters1.parameters());
+    // last value query
+    const data2 = await db.exec(filters2.applyQuery(sql2), filters2.parameters());
+
+    return { sums : data1, last : data2 }; // data1 and data2 are arrays
+  } catch (error) {
+    throw error();
+  }
+}
+
+
+async function finances(options) {
+  try {
+
+    // get the sum value for each column
+    const sql1 = `
+      SELECT 
+        SUM(IFNULL(fi.total_revenue, 0)) AS total_revenue
+        SUM(IFNULL(fi.total_subsidies, 0)) AS total_subsidies
+        SUM(IFNULL(fi.total_drugs_sale, 0)) AS total_drugs_sale,
+        SUM(IFNULL(fi.total_expenses, 0)) AS total_expenses,
+        SUM(IFNULL(fi.total_other_charge, 0)) AS total_other_charge,
+        SUM(IFNULL(fi.total_drugs_purchased, 0)) AS total_drugs_purchased,
+        SUM(IFNULL(fi.total_staff_charge, 0)) AS total_staff_charge,
+        SUM(IFNULL(fi.total_operating_charge, 0)) AS total_operating_charge,
+        SUM(IFNULL(fi.total_depreciation, 0)) AS total_depreciation,
+        SUM(IFNULL(fi.total_debts, 0)) AS total_debts,
+        SUM(IFNULL(fi.total_cash, 0)) AS total_cash,
+        SUM(IFNULL(fi.total_stock_value, 0)) AS total_stock_value,
+        SUM(IFNULL(fi.total_staff, 0)) AS total_staff,
+        p.start_date as period_start
+      FROM finances_indicator fi  
+      JOIN indicator ind ON ind.uuid = hi.indicator_uuid
+      JOIN period p ON p.id = ind.period_id
+    `;
+
+    db.convert(options, ['uuid', 'indicator_uuid']);
+
+    // this filter will be used for the sums query
+    let filters1 = new FilterParser(options, { tableAlias : 'fi' });
+
+    filters1 = defaultFilters(filters1);
+
+    // groupByPeriod key specify if it is necessary to group by period
+    if (options.groupByPeriod) {
+      filters1.setGroup('GROUP BY p.id');
+    }
+    // sums query
+    const data1 = await db.exec(filters1.applyQuery(sql1), filters1.parameters());
+
+    return { sums : data1 }; // data1 and data2 are arrays
+  } catch (error) {
+    throw error();
```

Comment:
Why is this error a function?  Also, do you need this try/catch?  Why not just pass the error up to the handling controller?